### DISCUSSION TITLE
Unable to access private gcr images even have right service account defined to scanjob

### DISCUSSION THREAD

**Discussion by saurabh21316** [aquasecurity/trivy#4625] [Bugs]
### Description

Hello Team,

I would really appreciate if someone can help to resolve the private gcr docker images issue.
ERROR:
{"level":"error","ts":"2023-06-08T16:35:25Z","logger":"reconciler.scan job","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-8644d5d6fc","container":"limit-processer","status.reason":"Error","status.message":"2023-06-08T16:35:21.135Z\t\u001b[31mFATAL\u001b[0m\timage scan error: scan error: unable to initialize a scanner: unable to initialize a docker scanner: 4 errors occurred:\n\t* unable to inspect the image (us.gcr.io/abc-docker-image/limit-processer:3ef762665895ce7ca28ddd33d77904d113a5451c): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\n\t* unable to initialize Podman client: no podman socket found: stat podman/podman.sock: no such file or directory\n\t* containerd socket not found: /run/containerd/containerd.sock\n\t* GET https://us.gcr.io/v2/abc-docker-image/limit-processer/manifests/3ef762665895ce7ca28ddd33d77904d113a5451c: UNAUTHORIZED: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication\n\n\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).processFailedScanJob\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:265\ngithub.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:79\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile/reconcile.go:103\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}

Two things I tried when deploying via helm:
1.) Via apply GCP service account directly to trivy-operator instead of using default sa.
serviceAccountName: abcserver
serviceAccount: abcserver.

2.) I also tried via apply annotation(gcp service account email id) to default trivy-operator sa.

None of it worked. 

When I deployed the below very basic pod config using same service account and it worked.

---
apiVersion: v1
kind: Pod
metadata:
  name: artifact-puller
spec:
  containers:
  - name: artifact-puller
    image: us.gcr.io/abc-docker-image/limit-processer:3ef762665895ce7ca28ddd33d77904d113a5451c
  serviceAccountName: trivy-operator
  nodeSelector:
    persistent-type: spot

so clearly issue with scan job or trivy operator, I believe.

I would highly appreciate if someone can point me to right direction.

Thanks

### Desired Behavior

shoulbe be able to scan the gcr images

### Actual Behavior

unable to scan it.

### Reproduction Steps

```bash
1.
2.
3.
...
```

### Target

Git Repository

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
ERROR:
{"level":"error","ts":"2023-06-08T16:35:25Z","logger":"reconciler.scan job","msg":"Scan job container","job":"trivy-system/scan-vulnerabilityreport-8644d5d6fc","container":"limit-processer","status.reason":"Error","status.message":"2023-06-08T16:35:21.135Z\t\u001b[31mFATAL\u001b[0m\timage scan error: scan error: unable to initialize a scanner: unable to initialize a docker scanner: 4 errors occurred:\n\t* unable to inspect the image (us.gcr.io/abc-docker-image/limit-processer:3ef762665895ce7ca28ddd33d77904d113a5451c): Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?\n\t* unable to initialize Podman client: no podman socket found: stat podman/podman.sock: no such file or directory\n\t* containerd socket not found: /run/containerd/containerd.sock\n\t* GET https://us.gcr.io/v2/abc-docker-image/limit-processer/manifests/3ef762665895ce7ca28ddd33d77904d113a5451c: UNAUTHORIZED: You don't have the needed permissions to perform this operation, and you may have invalid credentials. To authenticate your request, follow the steps in: https://cloud.google.com/container-registry/docs/advanced-authentication\n\n\n","stacktrace":"github.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).processFailedScanJob\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:265\ngithub.com/aquasecurity/trivy-operator/pkg/vulnerabilityreport/controller.(*ScanJobController).reconcileJobs.func1\n\t/home/runner/work/trivy-operator/trivy-operator/pkg/vulnerabilityreport/controller/scanjob.go:79\nsigs.k8s.io/controller-runtime/pkg/reconcile.Func.Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/reconcile/reconcile.go:103\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Reconcile\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:118\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).reconcileHandler\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:314\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).processNextWorkItem\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:265\nsigs.k8s.io/controller-runtime/pkg/internal/controller.(*Controller).Start.func2.2\n\t/home/runner/go/pkg/mod/sigs.k8s.io/controller-runtime@v0.15.0/pkg/internal/controller/controller.go:226"}
```

### Operating System

gke

### Version

```bash
0.14.0-rc
```

### Checklist

- [ ] Run `trivy --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by knqyf263** [comment]
Looks like it's a trivy-operator issue. Can you raise an issue in trivy-operator?
https://github.com/aquasecurity/trivy-operator
